src/lib/cache/gdfs_cache.hpp

```diff
@@ -94,9 +97,25 @@ class GDFSCache : public AbstractCacheImpl<Key, Value> {
     _map[key] = handle;
   }
 
-  Value& get(const Key& key) {
+  std::optional<Value> try_get(const Key& query) {
+    std::unique_lock<std::shared_mutex> lock(_mutex);
+    if (this->_capacity == 0) return std::nullopt;
+
+    auto it = _map.find(query);
+    if (it == _map.end()) return std::nullopt;
+
+    Handle handle = it->second;
+    GDFSCacheEntry& entry = (*handle);
+    entry.frequency++;
+    entry.priority = _inflation + static_cast<double>(entry.frequency) / entry.size;
+    _queue.update(handle);
+    return entry.value;
+  }
+
+  Value& get_entry(const Key& key) {
```

Comment:
Make `try_get` return an `optional<reference_wrapper<const Value>>` and simply assert that it is set here?

Also, why is it "get_entry" but just "try_get"? Rename to "get"?

Reply:
I guess the `optional` is for `try_get`, right?

I just used the names of the old cache class. `get` sounds better.

Reply:
Mmmh... good point. Forget about the reference_wrapper part above then.

Reply:
Okay, should I add a comment to the get method or make the Value& const?

Reply:
We cannot return a reference, even if it is const. Who guarantees that the referenced value will still be in the cache after we return?

That should be documented.